src/mito2/src/manifest/action.rs

```diff
@@ -84,12 +93,24 @@ pub enum TruncateKind {
 }
 
 /// The region manifest data.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct RegionManifest {
     /// Metadata of the region.
     pub metadata: RegionMetadataRef,
     /// SST files.
     pub files: HashMap<FileId, FileMeta>,
+    /// Removed files, which are not in the current manifest but may still be kept for a while.
+    /// This is a list of (set of files, timestamp) pairs, where the timestamp is the time when
+    /// the files are removed from manifest. The timestamp is in milliseconds since unix epoch.
+    ///
+    /// Using same checkpoint files and action files, the recovered manifest may differ in this
+    /// `removed_files` field, because the checkpointer may evict some removed files using
+    /// current machine time. This is acceptable because the removed files are not used in normal
+    /// read/write path.
+    ///
+    #[serde(default)]
+    pub removed_files: Vec<(HashSet<FileMeta>, i64)>,
```

Comment:
Better to define a struct. Can we only keep RegionId and FileId instead of the whole FileMeta to save some space?